ui/desktop/src/main.ts

```diff
@@ -1109,6 +1109,17 @@ ipcMain.on('react-ready', () => {
   console.log('[main] React ready - window is prepared for deep links');
 });
 
+// Handle external URL opening
+ipcMain.handle('open-external', async (_event, url: string) => {
```

Comment:
what functionality does this add that wasn't before - is it for cases where the type is links and it will open it in external browser? (if so - a hyperlink won't do that?)

Reply:
Since MCP-UI hyperlinks live inside of an iframe, clicking on a typical `<a href="" />` link will refresh the contents inside of the iframe. That might be perfect for some MCP-UIs but not others.

Also, I think that Electron brings in some form of protection that may prevent hyperlinks from performing as authored, to protect the app user. (I have a lack of knowledge in this area, I could be wrong) 

The MCP-UI SDK gives us the [link message type](https://mcpui.dev/guide/embeddable-ui#link) for, I believe, better security hygiene. So adding `open-external` was my attempt at finding the safest execution route.